pkg/pkgmgr/dpkg.go

```diff
@@ -459,115 +477,96 @@ func (dm *dpkgManager) unpackAndMergeUpdates(ctx context.Context, updates unvers
 			})).Root()
 	}
 
+	updated = updated.Run(buildkit.Sh("mkdir -p /tmp/debian-rootfs/var/lib/dpkg")).Root()
+
+	// Replace status file in tooling image with new status file with relevant pacakges from image to be patched.
+	// Regenerate /var/lib/dpkg/info files based on relevant pacakges from image to be patched.
+	dpkgdb := updated.Run(
+		llb.AddEnv("PACKAGES_PRESENT_ALL", string(jsonPackageData)),
+		llb.AddEnv("STATUS_FILE", dm.tempStatusFile),
+		llb.Args([]string{
+			`bash`, `-xec`, `
+							set -ex
+
+							json_str=$PACKAGES_PRESENT_ALL
+
+							rm -r /var/lib/dpkg/info
+							mkdir -p /var/lib/dpkg/info
+
+							apt-get update
+
+							while IFS=':' read -r package version; do
+								pkg_name=$(echo "$package" | sed 's/^"\(.*\)"$/\1/')
+								apt-get install --reinstall -y $pkg_name
+							done <<< "$(echo "$json_str" | tr -d '{}\n' | tr ',' '\n')"
+
+							apt --fix-broken install
+							dpkg --configure -a
+							apt-get check
+
+							echo "$STATUS_FILE" > /var/lib/dpkg/status
+						`,
+		})).Root()
```

Comment:
The var name bing captured here is `dpkgdb` but whats actually captured is the full rootfs, and then later on `/var/lib/dpkg` is copied out of it.

Maybe this should use `.AddMount("/var/lib/dpkg", updated, llb.SourcePath("/var/lib/dpkg")` instead of `.Root()`